src/lib/uri/uri.h

```diff
@@ -53,6 +61,21 @@ uri_destroy(struct uri *uri);
 int
 uri_format(char *str, int len, const struct uri *uri, bool write_password);
 
+/**
+ * Return @a uri query parameter value by given @a idx. If parameter with
+ * @a name does not exist or @a idx is greater than or equal to URI parameter
+ * value count, return NULL.
+ */
+const char *
+uri_query_param(const struct uri *uri, const char *name, int idx);
+
+/**
+ * Return count of values for @a uri query parameter with given @a name.
+ * If parameter with such @a name does not exist return 0.
+ */
+int
+uri_query_param_count(const struct uri *uri, const char *name);
```

Comment:
We discussed exposing `uri_query_param` with @alyapunov and decided that hiding it behind an API gives us more freedom when it comes to the implementation - e.g. we can replace the array storing parameters with a hash table without breaking uri users.

Re the name - I'm fine either way.

Reply:
Rename structure, but left it in the *.c file for the reason explained by @locker

Reply:
I don't see freedom here. Rather that it is quite complicated to access the parameters right now. Which "users" are you talking about? This C API is private. It is not available to users directly anyway.

Reply:
> Which "users" are you talking about?

Internal users.

> Maybe just expose `uri_query_param` to the header. It would be much easier than access the parameters via this function.

I don't see how this

```C
struct uri_param *param = uri_param(uri, "my_param");
for (int i = 0; i < param->value_count; i++) {
    const char *value = param->value[i];
}
```

is "much easier" than this

```C
for (int i = 0; i < uri_param_count(uri, "my_param"); i++) {
    const char *value = uri_param(uri, "my_param", i);
}
```

Bikeshed. Reworking it now would be a pointless setback IMO. I'm fine either way, but please ask @alyapunov, because he voted for the current variant for the reasons I tried to explain in this thread.